### ISSUE TITLE
Android - stuck on "Decrypting notes..." screen

### ISSUE THREAD

**Issue by thinklog** [standardnotes/forum#365]
UPDATE:
After experimenting a bit, I just discovered that if I log out before exiting the app, then I can log in again when the app restarts and then everything works fine, without uninstalling and reinstalling the app.
The behavior described below resulted from exiting the Standard Notes - for instance when shutting down the phone - without logging out. 
Therefore, my guess is that the behavior results from the apps inability to log in by itself again, once one has exited the app. When I described below that I had to uninstall and reinstall the app, all I was doing was forcing a new manual log in.
ORIGINAL POST
To access my notes I have to uninstall and reinstall Standard Notes after each use. This is uniformly so across all my Android devices. 
Immediately following installation everything works fine. If I leave the program and later restart it, it shows me a screen "Decrypting notes" and never finishes. (see attached screen shot.) 
This behavior began - I believe - when I tested out the FileSafe extension. I attached a small file using my Mac, but was not able view it on Android. I subsequently removed that file.
Since the trouble started I have removed the FileSafe extension altogether and reinstalled Standard Notes on all Android devices, but the problem remains. 
I hope you find a solution soon. It's tedious to constantly uninstall reinstall the program, whenever I want to review my shopping list in the supermarket (or work with other notes). Thanks.

![standard notes](https://user-images.githubusercontent.com/1121048/48156514-669d2280-e282-11e8-81e9-9bc9a88f3b28.png)

**Comment by mobitar** [comment]
Hmm the screenshot indicates that it's some difficulty loading the notes. I've seen this before in an older version of the Android app, but has since been fixed. Are you on the latest version by any chance? That would be v2.3.10.

**Comment by thinklog** [comment]
Version in the Settings screen says 2.3.10. Appears to be the latest version.

**Comment by mobitar** [comment]
Can you email me at mo@standardnotes.org from your account email? Want to double check your account and make sure everything looks healthy there. It's also possible that your phone may be getting stuck decrypting a very large note. If you're able to access the web or desktop version, and export a backup of your data from the Account menu, how large is that backup file?

**Comment by thinklog** [comment]
Done.

**Comment by monaxle** [comment]
I'm getting the same issue. Stuck on decrypting notes screen. v2.3.10 on Android.

**Comment by kevinclevenger** [comment]
+1 Android 9, SN 2.3.10. On new install first launch it flashes the notes list, then no notes are listed. With successive launches no notes are listed. 

When downgrading to SN 2.3.7 everything seems to work correctly.

**Comment by mobitar** [comment]
@kevinclevenger do you also see "Decrypting notes..." in the main view, or does it say "No notes."?

**Comment by thinklog** [comment]
Re Kevin: How did you downgrade? I'd like to do that too.

**Comment by kevinclevenger** [comment]
@mobitar Sorry, I don't recall. I'll try to retest.
@thinklog I use Yalp instead of Play Store, it can be configured to save the .apk after installation. You can also find older versions in third party apk repositories like [APK Pure](https://apkpure.com/standard-notes/com.standardnotes/versions)

**Comment by mobitar** [comment]
APKs for previous releases can be found here: https://github.com/standardnotes/mobile/releases

**Comment by thinklog** [comment]
Thanks. Just installed the previous version, and we're back in business.
Great

**Comment by mobitar** [comment]
Hmm, good to know that it's something to do with a recent release. What version did you install?

**Comment by thinklog** [comment]
Installed version 2.3.8

**Comment by monaxle** [comment]
I'm still getting stuck on the decrypting screen. SN 2.3.8 Android 7.1.2. Logging out before switching away from app avoids the need to uninstall then reinstall.

**Comment by monaxle** [comment]
Experimenting some more and I found this. SN v2.3.8 on Android 7. Leaving the app screen open with notes listed displayed disables the device lock when power button is pressed or phone is left unattended. Keeps SN open though! There does not seem to be a time limit to this. I left the phone over night in the above state and still able to access notes in the morning. Phone was still unlocked. Bit of a security issue.

**Comment by mobitar** [comment]
Thanks, will give these steps a go.

**Comment by mobitar** [comment]
This issue should be fixed in the 3.0 beta. Please give it a run here and see if that helps: https://play.google.com/apps/testing/com.standardnotes

**Comment by thinklog** [comment]
Wow! Standing ovation! Very nice. Works and it's a beauty.

**Comment by monaxle** [comment]
Thanks for the heads up on the beta. To save me a bit of time...

My first impression after installing: Wow! Standing ovation! Very nice. Works and it's a beauty. - **Agreed**

BUT: After exiting SN and switching to a different app and then restarting Standard Notes, it's stuck again on "Decrypting". - **shared experience**

Workaround: After tapping the Hamburger icon and selecting "Archived", then tapping the Hamburger icon again and selecting "All Notes", then all notes are displayed again. - **works for me too**

Interesting: If I modify a note on a different computer, and then force a screen refresh on the running instance that is frozen on "Decrypting ...", then the notes are displayed again. **works for me too**

In addition if I have the SN web app open the Android app does not get stuck on the decryption screen. This applies whether a note has been edited or not. 

In summary it seems like for now that the this iteration of the SN app for Android works well enough for me to use on my phone running on Android 7.1.2  

Thanks again. Great work.

**Comment by mobitar** [comment]
Thanks for the feedback, I'll investigate this more for the next beta.

**Comment by thinklog** [comment]
Looks great. Everything seems to be working! (That's 30 seconds after having installed it... ;-) But the previously reported problem no longer occurs on my device as of now.

**Comment by monaxle** [comment]
Beta 4 was more problematic than 3. The workarounds did not work around anymore. It then got better I guess with beta 5 and worked all the time without problems. Now on beta 6 and back to the issues with beta 4. I've had a lot on the past week and may well have missed other things along the way. Not very helpful I know.  

I can though with every iteration of the beta version tested sign out and then in again without uninstalling the app to bring all notes back again. The problem arises every time when swiping the app closed from recent items display and frequently (time related) when swiping it away with the back arrow on the Android home screen. I'll try and keep track of what's happening over the next day or so and report back.

**Comment by monaxle** [comment]
I will do. Do you recommend leaving the app installed and allowing it to update automatically when new beta is rolled out or uninstalling old version before installing new one?

**Comment by mobitar** [comment]
There's no need to uninstall between each version. Better not to actually, so you keep all your previous state and data without having to sign back in each time.

**Comment by monaxle** [comment]
Beta 7 gets stuck on decryption screen or only syncs notes from last couple of days. This happens when the app left signed in for a period of time. Can't say how long as I forget about it then when I go back the problem is there. Signing out then in again gets notes back. I don't plan to comment again until such time I experience something different. No comment means the problem persists.

**Comment by monaxle** [comment]
Woo yay! The SN app on my phone seems to be working now. This happened after updating the desktop app yesterday then signing out from my account and then signing back in again. Fingers crossed the issue I described above will not come back. I will update if it does. No update means it's all good.
